### ISSUE TITLE
Fat AAR generation

### ISSUE THREAD

**Issue by Roconda** [O2MC/moby-tracking-sdk#9]
A compiled AAR module doesn't link any dependencies statically by default. It would be desirable to have our dependencies statically linked into our distributed AAR module.

There are some projects that try to achieve this but all of them seem deprecated. 
For example: 

https://github.com/adwiv/android-fat-aar
https://stackoverflow.com/questions/49881928/android-create-uber-fat-aar-error-on-implementation
https://github.com/Vigi0303/fat-aar-plugin

**Comment by EdwinVanRooij** [comment]
It's not necessary to create a 'fat AAR'. Gradle handles dependencies intelligently.

Whenever there's one of the following lines in a `dependencies` block in a `build.gradle` file, Gradle will download the library on your filesystem:

- `compile 'com.company.project:name:1.0.0'` (deprecated, replace with `api`)
- `api 'com.company.project:name:1.0.0'` (compiles for current project and exposes the dependency for any module that uses this module (same behavior as `compile`))
- `implementation 'com.company.project:name:1.0.0'`: (compiles for the current project, won't expose for any module that uses this module, further information [here](https://medium.com/mindorks/implementation-vs-api-in-gradle-3-0-494c817a6fa))

Once it's downloaded, it's reused by any project that uses the same dependency. This ensures no unnecessary disk space usage. Any library that references another library will be sought for in the resulting repository. A simple practical example of this mechanism is to create a new app with the library [Picasso](https://github.com/square/picasso). Picasso uses [OkHttp](http://square.github.io/okhttp/) as well, however, it is not present in their [AAR](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.squareup.picasso%22%20AND%20a%3A%22picasso%22). Using Android Studio's 'go to declaration' feature, you'll find out that OkHttp is correctly referenced by Picasso.

This does not work for our SDK. Because we would use 
`implementation fileTree(dir: 'libs', include: ['*.aar'])`
to include the SDK into our sample app, gradle has not copied it into its cache repository. When we then reference the OkHttp library, Gradle will not look for the file in its cache repository. The result is an error indicating that the OkHttp domain classes can not be found. 
The fact that Gradle does not search through its cache repository seems like a bug/issue on their side.

The best solution seems to publish our SDK as an AAR to jCenter, just like Picasso did. Then, when referencing our SDK using the 'implementation' keyword, Gradle will put the SDK into it's cache repository. The OkHttp library will then be correctly referenced. 

A dirty workaround would be to copy OkHttp and other libraries locally into 'libs', then package 'libs' into the AAR like this:
```
task copyLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}
preBuild.dependsOn copyLibs

dependencies {
...
    implementation fileTree(dir: 'libs', include: ['*.jar'])
...
}
```

A quick test at this however reveals other issues like duplicate class/variable definitions from the android support library. Publishing the SDK to jCenter seems more desirable by far.

**Comment by EdwinVanRooij** [comment]
[Tutorial](https://android.jlelse.eu/publishing-your-android-kotlin-or-java-library-to-jcenter-from-android-studio-1b24977fe450) used to fix this.

**Comment by EdwinVanRooij** [comment]
Fixed as of #130.
